api/datasets.go

```diff
@@ -197,14 +198,69 @@ func extensionToMimeType(ext string) string {
 	}
 }
 
-func (h *DatasetHandlers) listHandler(w http.ResponseWriter, r *http.Request) {
-	args := lib.ListParamsFromRequest(r)
-	args.OrderBy = "created"
+// snoop reads from an io.ReadCloser and restores it so it can be read again
+func snoop(body *io.ReadCloser) (string, error) {
+	if body != nil && *body != nil {
+		result, err := ioutil.ReadAll(*body)
+		(*body).Close()
 
-	args.Term = r.FormValue("term")
+		if err != nil {
+			return "", err
+		}
 
-	res := []dsref.VersionInfo{}
-	if err := h.List(&args, &res); err != nil {
+		*body = ioutil.NopCloser(bytes.NewReader(result))
+		return string(result), nil
+	}
+	return "", nil
+}
+
+func parseProxyParams(r *http.Request, res interface{}) error {
```

Comment:
This is a bit overcomplicated but the point is getting to a generic path to parse the forwarded parameters.

Reply:
ok, I needed to think about this a bunch, and I think I'm answering a different question than this proxying project, but I wanted to deal with the issue of different forms of request parameter serialization:

https://github.com/qri-io/qri/commit/c8ff420471e05745222b1edbd6bfb5c352b7124b

The more I look at these new `Proxied` fields, the more I like the way we currently deal with RPC _within_ lib calls, and think part of this issue is just that the API is too thick